### ISSUE TITLE
Process SCRs for ALB for UAA and login to better control access

### ISSUE THREAD

**Issue by pburkholder** [cloud-gov/product#1426]
In order to better control access to login.* and uaa.* we need to implement a new ALB to front them, with customizable access rules. 

The endpoints uaa.fr.cloud.gov and login.fr.cloud.gov are both CloudFoundry apps associated with the *.fr.cloud.gov ALB.  Our plan had been to migrate all customers off of *.fr.cloud.gov so we can WAF-customize application paths with out any side effects or impacts.

However, in advance of that we can move those apps to an ALB specifically for `login.fr.cloud.gov` and `uaa.fr.cloud.gov` and those IPs will be resolved instead of the IPs for `*.fr.cloud.gov`.   With a WAF rule on that ALB we can control external access.

## Security considerations

- This will allow better customization of security measures.

## Implementation sketch:

- [ ] Deploy and test ALB in dev, then fr-stage, then fr

## Acceptance Criteria

- [ ]

## References

[DISCUSS & DECIDE - Better controlling access to  uaa.* and login.* - GitHub issue](https://github.com/cloud-gov/private/issues/170)

**Comment by karareinsel** [comment]
Needs to be complete by 10/30/2020.

**Comment by bengerman13** [comment]
I think this looks like:
- create a new ALB for uaa.$basedomain for each of our base domains (fr.cloud.gov, fr-stage.cloud.gov, etc). It should look basically exactly like the existing ALBs (wildcard cert, pointed at the gorouters, etc)
- test it, somehow. We could probably do this manually/ad-hoc by messing with hostfiles locally
- update DNS to point uaa.$basedomain at the new ALB. (Note that the old ALB will still work, and DNS can take time to converge, so if there are issues they may not show up immediately or consistently)

We should do this for each environment, and between finishing in staging and starting in production, we should run CATS

**Comment by pburkholder** [comment]
We should not do this in production until we have an OK from the JAB.

**Comment by pburkholder** [comment]
I've revised the SCR work I had in place earlier, could use [some pairing on the doc](https://docs.google.com/document/d/10YmiNE9W9F9lZzcRoQKx8fLFkshkko4wSizkwtH90mg/edit). Also drawing a diagram of how things fit together in a cg-diagrams draft PR (https://github.com/cloud-gov/cg-diagrams/pull/81).

**Comment by pburkholder** [comment]
[This SCR ](https://docs.google.com/document/d/10YmiNE9W9F9lZzcRoQKx8fLFkshkko4wSizkwtH90mg/edit) is ready for review and/or pairing. Let me know in Slack if you want to work on it, or just comment and accept in Google Docs.

**Comment by pburkholder** [comment]
@bengerman13 OK'd the updated SCR, so I've shipped that our JAB TRs. Now the soonest we'll hear back to 2/24, so marking accordingly.

**Comment by pburkholder** [comment]
Updated SCR to v3 and uploaded along with responses to their inquiries

**Comment by karareinsel** [comment]
Still waiting on answer from JAB.

**Comment by pburkholder** [comment]
This is still pending a decision by the JAB tech leads. Updating due date to 5/6

**Comment by pburkholder** [comment]
Update here from 5/12 meeting:

Nadine - what JAB TR’s are looking for
*  AWS managed rules don’t need to be assessed by a 3PAO
* Custom rules do need to be tested
* Peter - we’d use generic AWS rules for anything in front of customer workloads. Custom rules would be reserved for cloud.gov applications (anything that we’ve written and deployed).
* The implementation of WAF is designed to support cloud.gov customers. And any changes to custom rules for WAF are only for cloud.gov applications. 
* Since Coalfire is already working on the Pages SCR, Peter will see if we can do an amendment to the SAP to do the testing for AWS WAF.
* Protect against incidents. If you need to implement changes during an active incident, you can do that and address in the incident post-mortem. It falls under the incident communication procedure.

After that I initiated comms w/ CoalFire Federal.

They came back on 5/25 with:

> Our team is reaching out the PMO to confirm if a 3PAO assessment is required for this change. We should have more information shortly and will provide the LOE if needed. Additionally, after the Pages SCR the funded T&M bucket will have just over (redacted) available.

Then Joshua Scibelli replied on 5/28 -- but I was away for over a week for a family issue, and when I returned on 6/11 this went to the bottom of the stack. 

I'm replying now with context on what the JAB TRRs have asked for, and to clarify my confusion about the PMO vs the JAB TRRs.  I will include the cloud-gov-compliance list for reference and archives.

**Comment by pburkholder** [comment]
I'm getting this on the agenda with the JAB TR-Rs for tomorrow.

**Comment by pburkholder** [comment]
We are meeting w/ CoalFire Federal on Monday and if we can, we should have the ruleset list ready, the process (Terraform?) described, and a suggested testing methodology. That should be enough for them to scope work and, I hope, start it soon.

**Comment by pburkholder** [comment]
@kelleyconfer and I met w/ CFF today and they'll assemble an LOE, then we can more forward.

**Comment by pburkholder** [comment]
Updates: 

* CFF Anna says she'll provide LOE real soon now.
* JAB needs the SCR updated with our custom rulesets. @bengerman13 -- do you have an issue someplace with the custom rule to address that one weakness? Or can we write it up in https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/wafv2_regex_pattern_set format?

**Comment by pburkholder** [comment]
Our 3PAO scoped the LOE on this to 60 hours, which would take more of our team's time for an assessment that I can want to chew up. Trying to see if we can rescope into 2 SCRs.

**Comment by pburkholder** [comment]
Good outcome from today's meeting (see below). We should rescope this to three new tickets
- One for the new SCRs
- One for implementing the first SCR
- One for implementing the second SCR
-  
---
To recap that last part of our discussion today, we agreed that cloud.gov will withdraw the current WAF/ALB SCR. In its place, we'll submit two SCRs: 

- One for implementing the WAF with standard AWS rules, which is standard AWS tooling and will not require 3PAO assessment
- One for the custom WAF rules, which will be folded into our annual assessment.
We will proceed with submitting the first of these in the next couple of weeks.

**Comment by pburkholder** [comment]
> -- The JAB team concurs and approves your plan.  Thanks. 
🎉

**Comment by pburkholder** [comment]
[SCR cloud gov ALB+WAF core](https://docs.google.com/document/d/1GeWfj4x4zT7oKt4iaAEC5IADuD71-iQ82caTjAnbuSI/edit)

[SCR cloud gov ALB+WAF custom rules](https://docs.google.com/document/d/1FDTlMIqweYAvC4gbrih8mPG6GVgiZW23fW5dcbPcT0Y/edit)

**Comment by pburkholder** [comment]
I submitted both SCRs and emailed JAB TRs. I'll close this Thursday assuming it gets a thumbs up. Moving to blocked for now.

**Comment by pburkholder** [comment]
What's next here:
* Write up the process for how we roll custom WAF rules out to production
  * Testing
  * Review
  * Implementation
And then we'll submit that write up to 3PAO for an SCR review during annual assessment.

**Comment by ccostino** [comment]
@pburkholder, @soutenniza and I [created the write-up for this process](https://docs.google.com/document/d/1j47ZlaSegSaj89OKazaXiRHRux1eE_LeOlPsIleyb6I/edit#heading=h.pg6cnunuh71m) - please take a look and let us know if there's anything else you need or if there's anything we should adjust.  Thanks!

**Comment by pburkholder** [comment]
I created a [Google Doc folder](https://drive.google.com/drive/folders/1yUDq4tVJBSprFqigsrW47r481OhYwSeS) with the SCR and our write up, then shared it with CoalFire Federal.

**Comment by pburkholder** [comment]
I  think closing this is now just some email mopping up. I'm going to close it and carry on in  the implementation: https://github.com/cloud-gov/cg-provision/issues/912
